rfcs/20200713-benchdnn-ci-input-files/README.md

```diff
@@ -0,0 +1,95 @@
+# Update benchdnn input files for CI validation
+
+## Introduction
+
+Since amount of oneDNN functionality grows along with the necessary amount of
+testing coverage, CI testing time increases rapidly. In order to have meaningful
+functional testing with benchdnn, checking that it works and basic functionality
+is not broken for most popular compilers when new changes are introduced, there
+is a desire to separate general usability cases and implementation specific
+cases targeted performance (in terms of ISA branches, different memory formats,
+etc.) and move latter to the next level testing leaving general usability cases
+most frequently tested.
+
+In addition to this, current public CI, available for oneDNN external
+contributors, does not cover correctness testing properly since benchdnn
+validation is absent there. This proposal targets to close this gap as well.
+
+For internal oneDNN testing infrastructure re-considered and shortened CI
+coverage is expected to decrease machines load leading to higher testing cycles
+throughput.
+
+Current testing run time takes from 45 minutes and up to 20 hours, depending
+on various factors, such as OS, ISA, Engine and infrastructure stability and
+availability.
+
+## Proposal
+
+### Top level enabling bullets
+
+- CI inputs to cover HW-agnostic features meaning most library API abilities on
+  several relatively small shapes for all drivers.
+
+- Stop separating input files for CPU/GPU backend - library claims to provide
+  aligned support on both. All internal knowledge about unimplemented cases will
+  be hidden in benchdnn internals (like it's done for Gtests currently).
+
+- HW-specific implementations will utilize HW-agnostic input files whenever
+  possible, otherwise have additional dedicated input files.
```

Comment:
Do I understand correctly, that to test those separate files we will still have different targets in nightly (but keeping the CI input files the same)?

Reply:
Given that some time has passed and we've got some examples of such situation, it seems that a reasonable approach is to test only CI (light-weighted) inputs for side backends (whether CPU or GPU) and move the responsibility for a full functional testing on backend owners. CI coverage should secure there are no API or common part breaking changes which should enough to continue "lazy" maintenance of other backends besides Intel x64.

I have removed this bullet and replaced them with two from the top part of my comment, also answering your original question that following standard rules of CMake targets enabling and benchdnn file name convention is enough to proceed with a new input file.

Thanks for the review.